### ISSUE TITLE
upgrade dwv package

### ISSUE THREAD

**Issue by bradymiller** [openemr/openemr#1528]
Upgrade dwv package to 0.23:
-the architecture was revised to better use canvas when drawing.
-spring cleaning and moved all the demo viewers to their own repos
-Dependencies were revised to allow for better integration and a cleaner package.

**Comment by sjpadgett** [comment]
I'll take a look see and maybe can put together for upcoming release if it proves to add value in a noticeable way. Guess @MatthewVita might wish to weight in. I would really like to see us add ePDF and header parse to drag and drop or just parse out the patient data in the DICOM 10 to auto select patient and save to correct path. Could do in javascript and also will ensure we get the correct mime type regardless of O.S.

**Comment by bradymiller** [comment]
The upgrade is more maintenance stuff and placed this as a reminder for the future (ie. planned to do after the release). But the things you are talking about sound like good ideas for improved support. Am guessing the next thing we'll be also working on at some point is supporting scrolling through multi-image sets (such as a CT scan).

**Comment by MatthewVita** [comment]
> supporting scrolling through multi-image sets (such as a CT scan). Scroll does work!

**Comment by MatthewVita** [comment]
> I would really like to see us add ePDF and header parse to drag and drop

Are you talking in general here? I think Sherwin (?) added DnD in v5.0.1.

> DICOM 10 to auto select patient and save to correct path

So this would be in the "documents" area that is not coupled to a patient, right?

**Comment by sjpadgett** [comment]
Yep, DnD Sherwin added few months back and I test and use it all the time so, works well. My initial thoughts is yes to hook PnP, look at type, then present dialog with file supporting info for disposal. Something like that but for now I think it is important user can take existing DICOM received from radiology e.t.c and keep it with patient chart without export from their viewer.

**Comment by bradymiller** [comment]
Thanks @MatthewVita , I completely forgot that.

**Comment by MatthewVita** [comment]
@sjpadgett Sounds like a good idea. Feel free to run with it if you have the time/energy. I'll review and test, of course.
